### DISCUSSION TITLE
How do I use it without Visual Studio?

### DISCUSSION THREAD

**Discussion by zclllyybb** [microsoft/vcpkg#25440] [Q&A]
I just downloaded and configured vcpkg. But when I tried to install a package, I got this error message:

```
error: in triplet x64-windows: Unable to find a valid Visual Studio instance
Could not locate a complete Visual Studio instance
```

Does it mean it is necessary to install Visual Studio to make the vcpkg available?

Considering that all my work is doing on clion and vscode, if it's necessary, what's the minimum required components should I install?

Or maybe it's just my wrong somewhere?

**Comment by Osyotr** [comment]
Since you don't explicitly say what compiler you want to use, vcpkg (and CMake) defaults to latest MSVC on windows.
Solution: create custom triplet, set `VCPKG_CHAINLOAD_TOOLCHAIN_FILE` in it and use the same CMake toolchain that is used in CLion.
